docs/hugo/content/design/secrets.md

```diff
@@ -0,0 +1,670 @@
+
+**Should we drop the `Destination` suffix in the field names?**
+
+For example, the proposal for CosmosDB `DatabaseAccount` was:
+```yaml
+spec:
+  # Other spec fields elided...
+  forOperator:
+    operatorSpec:
+      keyDestination:
+        # stuff elided...
+      readOnlyKeyDestination:
+        # stuff elided...
+```
+
+Without the `Destination` suffix that would be:
+```yaml
+spec:
+  # Other spec fields elided...
+  operatorSpec:
+    secrets:
+      key:
+        # stuff elided...
+      readOnlyKey:
+        # stuff elided...
+```
+
+**Conclusion:** Yes, drop it
+
+**Do we allow writing a secret to a namespace where the resource isn't?**
+
+**Conclusion:** No, see: https://github.com/kubernetes/community/pull/5455
+
+**Do we support writing the same secret to multiple destinations?**
+
+Initially we had decided to not support this, and instead just write each secret to a maximum of 1 destination.
+This has one unfortunate side effect. It's tricky for customers to break apart different "parts" of the secret, such as
+the primary and secondary keys, into different secrets as the endpoint can only be written to one of them.
+
+There is a workaround for this though: when users want to split their primary/secondary keys into different secrets,
+they can write the endpoint to its own secret as well:
+```yaml
+spec:
+  # Other spec fields elided...
+  operatorSpec:
+     primaryKey:
+        type: Secret
+        name: my-secret
+        key: PRIMARY_KEY
+     secondaryKey:
+        type: Secret
+        name: my-secret
+        key: SECONDARY_KEY
+     readOnlyPrimaryKey:
+        type: Secret
+        name: my-readonly-secret
+        key: PRIMARY_KEY
+     readOnlySecondaryKey:
+        type: Secret
+        name: my-readonly-secret
+        key: SECONDARY_KEY
+     endpoint:
+        type: Secret
+        name: my-endpoint
+        key: ENDPOINT
+```
+
+Then if they want to use the secondary key for a pod they would mount the endpoint and the primary/secondary keys from
+`my-readonly-secret`.
+
+While this isn't quite as clean as it could otherwise be by supporting multiple secret destinations for each
+secret kind (`endpoint`, `readOnlyPrimaryKey`, etc) it's simpler to implement and has fewer failure modes as we don't
+have to deal with the possibility of the user specifying 10 destinations where we wrote some but failed to write others.
+
+**Conclusion:** No, we will not support this.
+
+**What client should we use for issuing the ListKeys/GetKeys request?**
+
+We don't automatically generate methods for performing these calls. It seems easiest to just use the corresponding Azure SDK for this.
+This does mean that we're taking a dependency on the SDK where we didn't have one before, but the overall footprint of usage is pretty small
+and it's going to be easier to do that than it is to write the methods ourselves.
+
+Conclusion: TBD
```

Comment:
```suggestion
**Conclusion**: TBD
```

Reply:
Good eye

Reply:
Actually we decided on this too. Will use the Azure SDK for it.